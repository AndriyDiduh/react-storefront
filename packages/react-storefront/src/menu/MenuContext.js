/**
 * @license
 * Copyright Â© 2017-2019 Moov Corporation.  All rights reserved.
 */
import React from 'react'
const MenuContext = React.createContext()
export default MenuContext
