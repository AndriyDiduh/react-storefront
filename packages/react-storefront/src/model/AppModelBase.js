/**
 * @license
 * Copyright Â© 2017-2018 Moov Corporation.  All rights reserved.
 */
import { types } from 'mobx-state-tree'
import { MenuModel } from '../Menu'
import { TabsModel } from '../NavTabs'
import CategoryModelBase from './CategoryModelBase'
import SubcategoryModelBase from './SubcategoryModelBase'
import ProductModelBase from './ProductModelBase'
import UserModelBase from './UserModelBase'
import CartModelBase from './CartModelBase'
import SearchModelBase from './SearchModelBase'
import isEqual from 'lodash/isEqual'

/**
 * Represents a single breadcrumb
 */
export const BreadcrumbModel = types.model('BreadcrumbModel', {
  /**
   * The URL to link to
   */
  url: types.maybeNull(types.string),

  /**
   * The text for the link
   */
  text: types.string,

  /**
   * An object to be applied to the state tree when the breadcrumb is clicked.  The shape should match your AppModel class.
   */
  state: types.frozen()
})

export const LocationModel = types
  .model({
    protocol: 'https',
    hostname: types.maybeNull(types.string),
    pathname: types.string,
    search: types.string,
    port: '443'
  })
  .views(self => ({
    get urlBase() {
      return (
        // Since protocols sometimes have a colon at the end, we remove it if it exists and add it back manually
        self.protocol.replace(/:/, '') +
        '://' +
        self.hostname +
        (['443', '80'].includes(self.port) ? '' : `:${self.port}`)
      )
    }
  }))

const AppModelBase = types
  .model('AppModelBase', {
    amp: false,
    offline: false,
    initialWidth: 'xs',
    page: types.maybeNull(types.string),
    title: types.maybeNull(types.string),
    description: types.maybeNull(types.string),
    loading: false,
    error: types.maybeNull(types.string),
    stack: types.maybeNull(types.string),
    menu: types.optional(MenuModel, {}),
    tabs: types.maybeNull(TabsModel),
    category: types.maybeNull(CategoryModelBase),
    subcategory: types.maybeNull(SubcategoryModelBase),
    product: types.maybeNull(ProductModelBase),
    loadingProduct: types.maybeNull(ProductModelBase),
    loadingSubcategory: types.maybeNull(SubcategoryModelBase),
    loadingCategory: types.maybeNull(CategoryModelBase),
    user: types.maybeNull(UserModelBase),
    location: types.maybeNull(LocationModel),
    search: types.optional(SearchModelBase, {}),
    breadcrumbs: types.optional(types.array(BreadcrumbModel), []),
    cart: types.optional(CartModelBase, {})
  })
  .views(self => ({
    get canonicalURL() {
      const { protocol, hostname, pathname, search } = self.location
      return protocol + '//' + hostname + pathname.replace(/\.amp$/, '') + search
    },
    get uri() {
      return self.location.pathname + self.location.search
    }
  }))
  .actions(self => ({
    /**
     * Clears the thumbnail being injected into the product skeleton
     */
    clearProductThumbnail() {
      self.productThumbnail = null
    },

    /**
     * Applies a patch to the state tree
     * @param {Object} patch The patch to apply
     * @param {String} action "PUSH", "POP", or "REPLACE"
     */
    applyState(patch, action) {
      if (action === 'POP') {
        // the user clicked the browser's back button
        patch = self.retainStateOnHistoryPop(patch) // ensure that state not corresponding to the URL is retained, for example the user and cart
        auditPatchOnPop(self, patch) // ensure that data for other pages is not changed.  This minimizes component reconciliation to boost performance

        if (patch.hasOwnProperty('page')) {
          self.page = patch.page // apply the page change first so the UI swaps to the previous page immediately
        }

        self.loading = false // if we're still loading the current page, we can hide the load mask immediately because we're going back to the previous page
        setImmediate(() => self.applyState(patch)) // apply the rest of the state change
      } else {
        // all other navigation
        const state = self.toJSON()

        for (let key in patch) {
          const value = patch[key]

          if (!isEqual(value, state[key])) {
            self[key] = value
          }
        }
      }
    },

    /**
     * Set connection status
     */
    setOffline(offline) {
      self.offline = offline
    },

    /**
     * Returns the part of the state tree which should not be overwritten
     * when the user goes forward or back.  You can override this action
     * to retain additional branches of the tree.
     */
    retainStateOnHistoryPop(patch) {
      delete patch.cart
      delete patch.user
      delete patch.menu
      delete patch.tabs
      return patch
    },

    /**
     * Sets the user model
     * @param {UserModelBase} user
     */
    signIn(user) {
      self.user = user
    },

    /**
     * Cleares the user model
     */
    signOut() {
      self.user = null
    },

    /**
     * Restores the user identity
     * @param {UserModelBase} user
     */
    setUser(user) {
      self.user = user
    },

    /**
     * Displays an error state
     * @param {Error} e
     */
    onError(e) {
      self.page = 'Error'
      self.error = e.message
      self.stack = e.stack
    }
  }))

/**
 * Removes values from the patch corresponding to existing values in the model that
 * implement forPage and for which forPage does not match the page being set in the patch.
 *
 * This helps minimize react component reconciliation by ensuring that values corresponding to
 * hidden pages are not changed when navigating back.
 *
 * @param {AppModelBase} model The app model instance
 * @param {Object} patch The patch to be applied
 */
function auditPatchOnPop(model, patch) {
  if (patch.page) {
    for (let key in patch) {
      const current = model[key]

      if (current && current.shouldApplyPatchOnPop && !current.shouldApplyPatchOnPop(patch)) {
        delete patch[key]
      }
    }
  }
}

export default AppModelBase
