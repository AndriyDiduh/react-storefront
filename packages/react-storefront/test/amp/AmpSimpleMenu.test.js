/**
 * @license
 * Copyright Â© 2017-2018 Moov Corporation.  All rights reserved.
 */
import React from 'react'
import { mount } from 'enzyme'
import Menu from '../../src/Menu'
import MenuContext from '../../src/menu/MenuContext'
import Provider from '../TestProvider'
import AppModelBase from '../../src/model/AppModelBase'

describe('AmpSimpleMenu', () => {
  let app

  beforeEach(() => {
    app = AppModelBase.create({
      amp: true,
      location: {
        pathname: '/',
        search: '',
        hostname: 'localhost'
      },
      menu: {
        levels: [
          {
            root: true,
            items: [
              {
                text: 'Group 1',
                className: 'group-1',
                items: [
                  {
                    text: 'Item 1',
                    url: '/item1',
                    items: [{ text: 'Child 1', url: '/item1/child1' }]
                  }
                ]
              },
              {
                text: 'Group 2',
                className: 'group-2',
                items: [
                  {
                    text: 'Item 2',
                    url: '/item2',
                    items: [{ text: 'Child 2', url: '/item2/child2' }]
                  }
                ]
              }
            ]
          }
        ]
      }
    })
  })

  it('should render simple AMP menu', () => {
    const wrapper = mount(
      <Provider app={app}>
        <Menu simple />
      </Provider>
    )

    expect(wrapper.find('AmpSimpleMenu').length).toBe(1)

    expect(wrapper.find('amp-accordion').length).toBe(4)
  })

  it('should match simple AMP menu snap', () => {
    const wrapper = mount(
      <Provider app={app}>
        <Menu simple />
      </Provider>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
