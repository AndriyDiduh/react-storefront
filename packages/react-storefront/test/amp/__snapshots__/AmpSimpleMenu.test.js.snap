// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AmpSimpleMenu should match simple AMP menu snap 1`] = `
<TestProvider app={{...}}>
  <Provider router={{...}} app={{...}} history={{...}} nextId={[Function: nextId]}>
    <MuiThemeProviderOld theme={{...}}>
      <WithTheme(WithStyles(inject-Menu-with-app)) simple={true}>
        <WithStyles(inject-Menu-with-app) theme={{...}} simple={true}>
          <inject-Menu-with-app theme={{...}} simple={true} classes={{...}}>
            <Menu theme={{...}} simple={true} classes={{...}} app={{...}} drawerWidth={330} expandFirstItem={false} align="left" trackSelected={false}>
              <WithTheme(WithStyles(inject-AmpSimpleMenu)) theme={{...}} simple={true} classes={{...}} app={{...}} drawerWidth={330} expandFirstItem={false} align="left" trackSelected={false}>
                <WithStyles(inject-AmpSimpleMenu) theme={{...}} simple={true} classes={{...}} app={{...}} drawerWidth={330} expandFirstItem={false} align="left" trackSelected={false}>
                  <inject-AmpSimpleMenu theme={{...}} simple={true} app={{...}} drawerWidth={330} expandFirstItem={false} align="left" trackSelected={false} classes={{...}}>
                    <AmpSimpleMenu theme={{...}} simple={true} app={{...}} drawerWidth={330} expandFirstItem={false} align="left" trackSelected={false} classes={{...}} menu={{...}} id="moov_menu">
                      <HelmetWrapper defer={true} encodeSpecialCharacters={true}>
                        <SideEffect(NullComponent) defer={true} encodeSpecialCharacters={true} script={{...}} style={{...}}>
                          <NullComponent defer={true} encodeSpecialCharacters={true} script={{...}} style={{...}} />
                        </SideEffect(NullComponent)>
                      </HelmetWrapper>
                      <amp-sidebar id="moov_menu" class="RSFAmpSimpleMenu-root-106" layout="nodisplay" side="left" on="sidebarClose:AMP.setState({ rsfMenu: { open: false } })">
                        <section>
                          <div>
                            <amp-accordion disable-session-states={true}>
                              <section>
                                <WithStyles(Typography) variant="body1" component="h3" className="RSFAmpSimpleMenu-item-107 RSFAmpSimpleMenu-group-108 group-1">
                                  <Typography variant="body1" component="h3" className="RSFAmpSimpleMenu-item-107 RSFAmpSimpleMenu-group-108 group-1" theme={{...}} classes={{...}} align="inherit" color="default" gutterBottom={false} headlineMapping={{...}} noWrap={false} paragraph={false}>
                                    <h3 className="MuiTypography-root-112 MuiTypography-body1-121 RSFAmpSimpleMenu-item-107 RSFAmpSimpleMenu-group-108 group-1">
                                      Group 1
                                      <pure(ExpandLessIcon) className="RSFAmpSimpleMenu-toggle-109 RSFAmpSimpleMenu-collapse-111">
                                        <ExpandLessIcon className="RSFAmpSimpleMenu-toggle-109 RSFAmpSimpleMenu-collapse-111">
                                          <WithStyles(SvgIcon) className="RSFAmpSimpleMenu-toggle-109 RSFAmpSimpleMenu-collapse-111">
                                            <SvgIcon className="RSFAmpSimpleMenu-toggle-109 RSFAmpSimpleMenu-collapse-111" classes={{...}} color="inherit" component="svg" fontSize="default" viewBox="0 0 24 24">
                                              <svg className="MuiSvgIcon-root-147 RSFAmpSimpleMenu-toggle-109 RSFAmpSimpleMenu-collapse-111" focusable="false" viewBox="0 0 24 24" color={[undefined]} aria-hidden="true" role="presentation">
                                                <path d="M12 8l-6 6 1.41 1.41L12 10.83l4.59 4.58L18 14z" />
                                                <path fill="none" d="M0 0h24v24H0z" />
                                              </svg>
                                            </SvgIcon>
                                          </WithStyles(SvgIcon)>
                                        </ExpandLessIcon>
                                      </pure(ExpandLessIcon)>
                                      <pure(ExpandMoreIcon) className="RSFAmpSimpleMenu-toggle-109 RSFAmpSimpleMenu-expand-110">
                                        <ExpandMoreIcon className="RSFAmpSimpleMenu-toggle-109 RSFAmpSimpleMenu-expand-110">
                                          <WithStyles(SvgIcon) className="RSFAmpSimpleMenu-toggle-109 RSFAmpSimpleMenu-expand-110">
                                            <SvgIcon className="RSFAmpSimpleMenu-toggle-109 RSFAmpSimpleMenu-expand-110" classes={{...}} color="inherit" component="svg" fontSize="default" viewBox="0 0 24 24">
                                              <svg className="MuiSvgIcon-root-147 RSFAmpSimpleMenu-toggle-109 RSFAmpSimpleMenu-expand-110" focusable="false" viewBox="0 0 24 24" color={[undefined]} aria-hidden="true" role="presentation">
                                                <path d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z" />
                                                <path fill="none" d="M0 0h24v24H0z" />
                                              </svg>
                                            </SvgIcon>
                                          </WithStyles(SvgIcon)>
                                        </ExpandMoreIcon>
                                      </pure(ExpandMoreIcon)>
                                    </h3>
                                  </Typography>
                                </WithStyles(Typography)>
                                <div>
                                  <amp-accordion disable-session-states={true}>
                                    <section>
                                      <WithStyles(Typography) variant="body1" component="h3" className="RSFAmpSimpleMenu-item-107 RSFAmpSimpleMenu-group-108">
                                        <Typography variant="body1" component="h3" className="RSFAmpSimpleMenu-item-107 RSFAmpSimpleMenu-group-108" theme={{...}} classes={{...}} align="inherit" color="default" gutterBottom={false} headlineMapping={{...}} noWrap={false} paragraph={false}>
                                          <h3 className="MuiTypography-root-112 MuiTypography-body1-121 RSFAmpSimpleMenu-item-107 RSFAmpSimpleMenu-group-108">
                                            Item 1
                                            <pure(ExpandLessIcon) className="RSFAmpSimpleMenu-toggle-109 RSFAmpSimpleMenu-collapse-111">
                                              <ExpandLessIcon className="RSFAmpSimpleMenu-toggle-109 RSFAmpSimpleMenu-collapse-111">
                                                <WithStyles(SvgIcon) className="RSFAmpSimpleMenu-toggle-109 RSFAmpSimpleMenu-collapse-111">
                                                  <SvgIcon className="RSFAmpSimpleMenu-toggle-109 RSFAmpSimpleMenu-collapse-111" classes={{...}} color="inherit" component="svg" fontSize="default" viewBox="0 0 24 24">
                                                    <svg className="MuiSvgIcon-root-147 RSFAmpSimpleMenu-toggle-109 RSFAmpSimpleMenu-collapse-111" focusable="false" viewBox="0 0 24 24" color={[undefined]} aria-hidden="true" role="presentation">
                                                      <path d="M12 8l-6 6 1.41 1.41L12 10.83l4.59 4.58L18 14z" />
                                                      <path fill="none" d="M0 0h24v24H0z" />
                                                    </svg>
                                                  </SvgIcon>
                                                </WithStyles(SvgIcon)>
                                              </ExpandLessIcon>
                                            </pure(ExpandLessIcon)>
                                            <pure(ExpandMoreIcon) className="RSFAmpSimpleMenu-toggle-109 RSFAmpSimpleMenu-expand-110">
                                              <ExpandMoreIcon className="RSFAmpSimpleMenu-toggle-109 RSFAmpSimpleMenu-expand-110">
                                                <WithStyles(SvgIcon) className="RSFAmpSimpleMenu-toggle-109 RSFAmpSimpleMenu-expand-110">
                                                  <SvgIcon className="RSFAmpSimpleMenu-toggle-109 RSFAmpSimpleMenu-expand-110" classes={{...}} color="inherit" component="svg" fontSize="default" viewBox="0 0 24 24">
                                                    <svg className="MuiSvgIcon-root-147 RSFAmpSimpleMenu-toggle-109 RSFAmpSimpleMenu-expand-110" focusable="false" viewBox="0 0 24 24" color={[undefined]} aria-hidden="true" role="presentation">
                                                      <path d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z" />
                                                      <path fill="none" d="M0 0h24v24H0z" />
                                                    </svg>
                                                  </SvgIcon>
                                                </WithStyles(SvgIcon)>
                                              </ExpandMoreIcon>
                                            </pure(ExpandMoreIcon)>
                                          </h3>
                                        </Typography>
                                      </WithStyles(Typography)>
                                      <div>
                                        <WithStyles(inject-Link) className="RSFAmpSimpleMenu-item-107" to="/item1/child1">
                                          <inject-Link className="RSFAmpSimpleMenu-item-107" to="/item1/child1" classes={{...}}>
                                            <Link className="RSFAmpSimpleMenu-item-107" to="/item1/child1" classes={{...}} history={{...}} router={{...}} app={{...}} anchorProps={{...}} preserveScroll={false}>
                                              <a data-moov-link="on" className="RSFLink-root-156 RSFAmpSimpleMenu-item-107" style={[undefined]} onClick={[Function]} href="https://localhost/item1/child1">
                                                <WithStyles(ListItemText)>
                                                  <ListItemText classes={{...}} disableTypography={false} inset={false}>
                                                    <div className="MuiListItemText-root-157">
                                                      <WithStyles(Typography) variant="subheading" internalDeprecatedVariant={true} className="MuiListItemText-primary-160" component="span">
                                                        <Typography variant="subheading" internalDeprecatedVariant={true} className="MuiListItemText-primary-160" component="span" theme={{...}} classes={{...}} align="inherit" color="default" gutterBottom={false} headlineMapping={{...}} noWrap={false} paragraph={false}>
                                                          <span className="MuiTypography-root-112 MuiTypography-subtitle1-130 MuiListItemText-primary-160">
                                                            Child 1
                                                          </span>
                                                        </Typography>
                                                      </WithStyles(Typography)>
                                                    </div>
                                                  </ListItemText>
                                                </WithStyles(ListItemText)>
                                              </a>
                                            </Link>
                                          </inject-Link>
                                        </WithStyles(inject-Link)>
                                      </div>
                                    </section>
                                  </amp-accordion>
                                </div>
                              </section>
                            </amp-accordion>
                            <amp-accordion disable-session-states={true}>
                              <section>
                                <WithStyles(Typography) variant="body1" component="h3" className="RSFAmpSimpleMenu-item-107 RSFAmpSimpleMenu-group-108 group-2">
                                  <Typography variant="body1" component="h3" className="RSFAmpSimpleMenu-item-107 RSFAmpSimpleMenu-group-108 group-2" theme={{...}} classes={{...}} align="inherit" color="default" gutterBottom={false} headlineMapping={{...}} noWrap={false} paragraph={false}>
                                    <h3 className="MuiTypography-root-112 MuiTypography-body1-121 RSFAmpSimpleMenu-item-107 RSFAmpSimpleMenu-group-108 group-2">
                                      Group 2
                                      <pure(ExpandLessIcon) className="RSFAmpSimpleMenu-toggle-109 RSFAmpSimpleMenu-collapse-111">
                                        <ExpandLessIcon className="RSFAmpSimpleMenu-toggle-109 RSFAmpSimpleMenu-collapse-111">
                                          <WithStyles(SvgIcon) className="RSFAmpSimpleMenu-toggle-109 RSFAmpSimpleMenu-collapse-111">
                                            <SvgIcon className="RSFAmpSimpleMenu-toggle-109 RSFAmpSimpleMenu-collapse-111" classes={{...}} color="inherit" component="svg" fontSize="default" viewBox="0 0 24 24">
                                              <svg className="MuiSvgIcon-root-147 RSFAmpSimpleMenu-toggle-109 RSFAmpSimpleMenu-collapse-111" focusable="false" viewBox="0 0 24 24" color={[undefined]} aria-hidden="true" role="presentation">
                                                <path d="M12 8l-6 6 1.41 1.41L12 10.83l4.59 4.58L18 14z" />
                                                <path fill="none" d="M0 0h24v24H0z" />
                                              </svg>
                                            </SvgIcon>
                                          </WithStyles(SvgIcon)>
                                        </ExpandLessIcon>
                                      </pure(ExpandLessIcon)>
                                      <pure(ExpandMoreIcon) className="RSFAmpSimpleMenu-toggle-109 RSFAmpSimpleMenu-expand-110">
                                        <ExpandMoreIcon className="RSFAmpSimpleMenu-toggle-109 RSFAmpSimpleMenu-expand-110">
                                          <WithStyles(SvgIcon) className="RSFAmpSimpleMenu-toggle-109 RSFAmpSimpleMenu-expand-110">
                                            <SvgIcon className="RSFAmpSimpleMenu-toggle-109 RSFAmpSimpleMenu-expand-110" classes={{...}} color="inherit" component="svg" fontSize="default" viewBox="0 0 24 24">
                                              <svg className="MuiSvgIcon-root-147 RSFAmpSimpleMenu-toggle-109 RSFAmpSimpleMenu-expand-110" focusable="false" viewBox="0 0 24 24" color={[undefined]} aria-hidden="true" role="presentation">
                                                <path d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z" />
                                                <path fill="none" d="M0 0h24v24H0z" />
                                              </svg>
                                            </SvgIcon>
                                          </WithStyles(SvgIcon)>
                                        </ExpandMoreIcon>
                                      </pure(ExpandMoreIcon)>
                                    </h3>
                                  </Typography>
                                </WithStyles(Typography)>
                                <div>
                                  <amp-accordion disable-session-states={true}>
                                    <section>
                                      <WithStyles(Typography) variant="body1" component="h3" className="RSFAmpSimpleMenu-item-107 RSFAmpSimpleMenu-group-108">
                                        <Typography variant="body1" component="h3" className="RSFAmpSimpleMenu-item-107 RSFAmpSimpleMenu-group-108" theme={{...}} classes={{...}} align="inherit" color="default" gutterBottom={false} headlineMapping={{...}} noWrap={false} paragraph={false}>
                                          <h3 className="MuiTypography-root-112 MuiTypography-body1-121 RSFAmpSimpleMenu-item-107 RSFAmpSimpleMenu-group-108">
                                            Item 2
                                            <pure(ExpandLessIcon) className="RSFAmpSimpleMenu-toggle-109 RSFAmpSimpleMenu-collapse-111">
                                              <ExpandLessIcon className="RSFAmpSimpleMenu-toggle-109 RSFAmpSimpleMenu-collapse-111">
                                                <WithStyles(SvgIcon) className="RSFAmpSimpleMenu-toggle-109 RSFAmpSimpleMenu-collapse-111">
                                                  <SvgIcon className="RSFAmpSimpleMenu-toggle-109 RSFAmpSimpleMenu-collapse-111" classes={{...}} color="inherit" component="svg" fontSize="default" viewBox="0 0 24 24">
                                                    <svg className="MuiSvgIcon-root-147 RSFAmpSimpleMenu-toggle-109 RSFAmpSimpleMenu-collapse-111" focusable="false" viewBox="0 0 24 24" color={[undefined]} aria-hidden="true" role="presentation">
                                                      <path d="M12 8l-6 6 1.41 1.41L12 10.83l4.59 4.58L18 14z" />
                                                      <path fill="none" d="M0 0h24v24H0z" />
                                                    </svg>
                                                  </SvgIcon>
                                                </WithStyles(SvgIcon)>
                                              </ExpandLessIcon>
                                            </pure(ExpandLessIcon)>
                                            <pure(ExpandMoreIcon) className="RSFAmpSimpleMenu-toggle-109 RSFAmpSimpleMenu-expand-110">
                                              <ExpandMoreIcon className="RSFAmpSimpleMenu-toggle-109 RSFAmpSimpleMenu-expand-110">
                                                <WithStyles(SvgIcon) className="RSFAmpSimpleMenu-toggle-109 RSFAmpSimpleMenu-expand-110">
                                                  <SvgIcon className="RSFAmpSimpleMenu-toggle-109 RSFAmpSimpleMenu-expand-110" classes={{...}} color="inherit" component="svg" fontSize="default" viewBox="0 0 24 24">
                                                    <svg className="MuiSvgIcon-root-147 RSFAmpSimpleMenu-toggle-109 RSFAmpSimpleMenu-expand-110" focusable="false" viewBox="0 0 24 24" color={[undefined]} aria-hidden="true" role="presentation">
                                                      <path d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z" />
                                                      <path fill="none" d="M0 0h24v24H0z" />
                                                    </svg>
                                                  </SvgIcon>
                                                </WithStyles(SvgIcon)>
                                              </ExpandMoreIcon>
                                            </pure(ExpandMoreIcon)>
                                          </h3>
                                        </Typography>
                                      </WithStyles(Typography)>
                                      <div>
                                        <WithStyles(inject-Link) className="RSFAmpSimpleMenu-item-107" to="/item2/child2">
                                          <inject-Link className="RSFAmpSimpleMenu-item-107" to="/item2/child2" classes={{...}}>
                                            <Link className="RSFAmpSimpleMenu-item-107" to="/item2/child2" classes={{...}} history={{...}} router={{...}} app={{...}} anchorProps={{...}} preserveScroll={false}>
                                              <a data-moov-link="on" className="RSFLink-root-156 RSFAmpSimpleMenu-item-107" style={[undefined]} onClick={[Function]} href="https://localhost/item2/child2">
                                                <WithStyles(ListItemText)>
                                                  <ListItemText classes={{...}} disableTypography={false} inset={false}>
                                                    <div className="MuiListItemText-root-157">
                                                      <WithStyles(Typography) variant="subheading" internalDeprecatedVariant={true} className="MuiListItemText-primary-160" component="span">
                                                        <Typography variant="subheading" internalDeprecatedVariant={true} className="MuiListItemText-primary-160" component="span" theme={{...}} classes={{...}} align="inherit" color="default" gutterBottom={false} headlineMapping={{...}} noWrap={false} paragraph={false}>
                                                          <span className="MuiTypography-root-112 MuiTypography-subtitle1-130 MuiListItemText-primary-160">
                                                            Child 2
                                                          </span>
                                                        </Typography>
                                                      </WithStyles(Typography)>
                                                    </div>
                                                  </ListItemText>
                                                </WithStyles(ListItemText)>
                                              </a>
                                            </Link>
                                          </inject-Link>
                                        </WithStyles(inject-Link)>
                                      </div>
                                    </section>
                                  </amp-accordion>
                                </div>
                              </section>
                            </amp-accordion>
                          </div>
                        </section>
                      </amp-sidebar>
                    </AmpSimpleMenu>
                  </inject-AmpSimpleMenu>
                </WithStyles(inject-AmpSimpleMenu)>
              </WithTheme(WithStyles(inject-AmpSimpleMenu))>
            </Menu>
          </inject-Menu-with-app>
        </WithStyles(inject-Menu-with-app)>
      </WithTheme(WithStyles(inject-Menu-with-app))>
    </MuiThemeProviderOld>
  </Provider>
</TestProvider>
`;
